cmd/pathUtils.go

```diff
@@ -119,8 +109,93 @@ func determineLocationLevel(location string, locationType common.Location, sourc
 	}
 }
 
-// Both of the below functions only really do one thing at the moment.
+// ----- ROOT PATH GRABBING -----
+
+// resourceBase will always be returned regardless of the location.
+// resourceToken will be separated and returned depending on the location.
+func SplitAuthTokenFromResource(resource string, location common.Location) (resourceBase, resourceToken string, err error) {
+	switch location {
+	case common.ELocation.Local(),
+		common.ELocation.S3(),
+		common.ELocation.Benchmark(), // cover for benchmark as we generate data for that
+		common.ELocation.Unknown():   // cover for unknown as we treat that as garbage
+		// Local and S3 don't feature URL-embedded tokens
+		return resource, "", nil
+
+	// Use resource-specific APIs that all mostly do the same thing, just on the off-chance they end up doing something slightly different in the future.
+	// TODO: make GetAccountRoot and GetContainerName use their own specific APIs as well. It's _unlikely_ at best that the URL format will change drastically.
+	//       but just on the off-chance that it does, I'd prefer if AzCopy could adapt adequately as soon as the SDK catches the change
```

Comment:
Good thinking with these TODOs. And I agree that they are TODOs (i.e. shouldn't be done now).  For the first one, maybe add a backlog item, since it sounds like something that may catch us out again if we don't fix it. (Where's the second one, about find a clever way to reduce code duplication) only merits an in-code TODO, and not any backlog item).